Starting marytts-server fails

Hi folks,

this happens with the version from git from Monday:

I start the server like 
```
sh marytts-server
```

This is the output in the console:
```
MARY server 5.2-SNAPSHOT starting as a HTTP server...Exception in thread "main" marytts.exceptions.MaryConfigurationException: Cannot instantiate object from 'marytts.language.de.JPhonemiser': 1
	at marytts.util.MaryRuntimeUtils.instantiateObject(MaryRuntimeUtils.java:135)
	at marytts.modules.ModuleRegistry.instantiateModule(ModuleRegistry.java:87)
	at marytts.server.Mary.startModules(Mary.java:130)
	at marytts.server.Mary.startup(Mary.java:291)
	at marytts.server.Mary.startup(Mary.java:201)
	at marytts.server.Mary.main(Mary.java:491)
Caused by: java.lang.ArrayIndexOutOfBoundsException: 1
	at marytts.modules.JPhonemiser.readLexicon(JPhonemiser.java:369)
	at marytts.modules.JPhonemiser.<init>(JPhonemiser.java:109)
	at marytts.modules.JPhonemiser.<init>(JPhonemiser.java:87)
	at marytts.language.de.JPhonemiser.<init>(JPhonemiser.java:73)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:57)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:526)
	at java.lang.Class.newInstance(Class.java:383)
	at marytts.util.MaryRuntimeUtils.instantiateObject(MaryRuntimeUtils.java:131)
	... 5 more
Exception in thread "Thread-1" java.lang.IllegalStateException: MARY system is not running
	at marytts.server.Mary.shutdown(Mary.java:363)
	at marytts.server.Mary$3.run(Mary.java:284)
```

This is from the server.log:

2015-08-05 13:22:00,983 [main] INFO  marytts.main Mary starting up...
2015-08-05 13:22:00,983 [main] INFO  marytts.main Specification version 5.2-SNAPSHOT
2015-08-05 13:22:00,983 [main] INFO  marytts.main Implementation version unknown
2015-08-05 13:22:00,983 [main] INFO  marytts.main Running on a Java 1.7.0_85 implementation by Oracle Corporation, on a Linux platform (amd64, 2.6.32-504.23.4.el6.x86_64)
2015-08-05 13:22:01,217 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.language.en.JTokeniser'
2015-08-05 13:22:01,221 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.language.en.Preprocess'
2015-08-05 13:22:01,369 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.modules.JPhonemiser(en_US.)'
2015-08-05 13:22:01,457 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.modules.JPhonemiser(en_GB.)'
2015-08-05 13:22:01,515 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.language.en.Prosody'
2015-08-05 13:22:01,518 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.modules.SimplePhoneme2AP(en_US)'
2015-08-05 13:22:01,519 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.language.en.PronunciationModel'
2015-08-05 13:22:01,520 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.modules.OpenNLPPosTagger(en,en.pos)'
2015-08-05 13:22:01,520 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.modules.JPhonemiser(sv.)'
2015-08-05 13:22:01,538 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.modules.MinimalisticPosTagger(sv,sv.)'
2015-08-05 13:22:01,539 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.language.it.JTokeniser'
2015-08-05 13:22:01,539 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.language.it.Preprocess'
2015-08-05 13:22:01,540 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.modules.JPhonemiser(it.)'
2015-08-05 13:22:01,564 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.modules.OpenNLPPosTagger(it,it.pos)'
2015-08-05 13:22:01,564 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.language.it.Prosody'
2015-08-05 13:22:01,565 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.language.it.Postlex'
2015-08-05 13:22:01,565 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.modules.Synthesis'
2015-08-05 13:22:01,566 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.modules.TextToMaryXML'
2015-08-05 13:22:01,567 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.modules.SableParser'
2015-08-05 13:22:01,567 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.modules.SSMLParser'
2015-08-05 13:22:01,568 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.modules.APMLParser'
2015-08-05 13:22:01,568 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.modules.EmotionmlParser'
2015-08-05 13:22:01,569 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.modules.JTokeniser'
2015-08-05 13:22:01,569 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.modules.DummyTokens2Words'
2015-08-05 13:22:01,569 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.modules.ProsodyGeneric'
2015-08-05 13:22:01,569 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.modules.PronunciationModel'
2015-08-05 13:22:01,569 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.modules.TargetFeatureLister'
2015-08-05 13:22:01,570 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.modules.HalfPhoneTargetFeatureLister'
2015-08-05 13:22:01,570 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.modules.AcousticModeller'
2015-08-05 13:22:01,571 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.modules.RealisedAcoustparamsExtractor'
2015-08-05 13:22:01,571 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.modules.RealisedDurationsExtractor'
2015-08-05 13:22:01,572 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.modules.HTSEngine'
2015-08-05 13:22:01,573 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.modules.PraatTextGridGenerator'
2015-08-05 13:22:01,574 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.language.te.JTokeniser'
2015-08-05 13:22:01,574 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.modules.DummyTokens2Words'
2015-08-05 13:22:01,574 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.language.te.JPhonemiser(te.)'
2015-08-05 13:22:01,577 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.modules.ProsodyGeneric'
2015-08-05 13:22:01,577 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.modules.MinimalisticPosTagger(te,te)'
2015-08-05 13:22:01,577 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.language.ru.Tokenizer'
2015-08-05 13:22:01,577 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.modules.DummyTokens2Words'
2015-08-05 13:22:01,577 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.modules.JPhonemiser(ru.)'
2015-08-05 13:22:01,751 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.modules.ProsodyGeneric'
2015-08-05 13:22:01,751 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.modules.MinimalisticPosTagger(ru,ru)'
2015-08-05 13:22:01,751 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.language.de.JTokeniser'
2015-08-05 13:22:01,752 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.language.de.Preprocess'
2015-08-05 13:22:01,752 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.modules.OpenNLPPosTagger(de,de.pos)'
2015-08-05 13:22:01,753 [main] INFO  marytts.ModuleRegistry Now initiating mary module 'marytts.language.de.JPhonemiser'

With an older version of marytts in a different location, everything works fine...


Who can help?

Thanks!

--- a/marytts-runtime/src/main/java/marytts/modules/JPhonemiser.java
+++ b/marytts-runtime/src/main/java/marytts/modules/JPhonemiser.java
@@ -355,6 +355,7 @@ public class JPhonemiser extends InternalModule {
 	 * @return
 	 */
 	protected Map<String, List<String>> readLexicon(String lexiconFilename) throws IOException {
+		logger.debug(String.format("Reading lexicon from '%s'", lexiconFilename));
 		String line;
 		Map<String, List<String>> fLexicon = new HashMap<String, List<String>>();
 
@@ -366,11 +367,18 @@ public class JPhonemiser extends InternalModule {
 
 			String[] lineParts = line.split("\\s*\\|\\s*");
 			String graphStr = lineParts[0];
-			String phonStr = lineParts[1];
+			String phonStr = null;
+			try {
+				phonStr = lineParts[1];
+			} catch (ArrayIndexOutOfBoundsException e) {
+				logger.warn(String.format("Lexicon '%s': missing transcription for '%s'", lexiconFilename, graphStr));
+				continue;
+			}
 			try {
 				allophoneSet.splitIntoAllophones(phonStr);
-			} catch (RuntimeException re) {
-				logger.warn("Lexicon '" + lexiconFilename + "': invalid entry for '" + graphStr + "'", re);
+			} catch (IllegalArgumentException e) {
+				logger.warn(String.format("Lexicon '%s': invalid entry for '%s': %s", lexiconFilename, graphStr, e.getMessage()));
+				continue;
 			}
 			String phonPosStr = phonStr;
 			if (lineParts.length > 2) {

